rdtools/test/filtering_test.py

```diff
@@ -156,10 +157,12 @@ def test_logic_clip_filter(generate_power_time_series_no_clipping,
                   power_datetime_index_nc[:9])
     # Test that a warning is thrown when the time series is tz-naive
     warnings.simplefilter("always")
-    with warnings.catch_warnings(record=True) as w:
+    with warnings.catch_warnings(record=True) as record:
         logic_clip_filter(power_nc_tz_naive)
         # Warning thrown for it being an experimental filter + tz-naive
-        assert len(w) == 2
+        assert_warnings(['The logic-based filter is an experimental',
+                         'Function expects timestamps in local time'],
+                        record)
```

Comment:
These warning changes aren't related to py 3.10 or 3.6.  It's just because testing the number of emitted warnings is rather brittle and failed on the new 3.10 test because of some unrelated warning from an indirect dependency.  Normally I'd used `pytest.warns` to check the specific warning message text, but I don't think it has a way to check for multiple warnings like we need to do here.  Thus the new function `assert_warnings` in `conftest.py`.